locator/topology.cc

```diff
@@ -426,7 +427,7 @@ const node* topology::find_node(const inet_address& ep) const noexcept {
 // Finds a node by its index
 // Returns nullptr if not found
 const node* topology::find_node(node::idx_type idx) const noexcept {
-    if (idx >= _nodes.size()) {
+    if (std::cmp_greater_equal(idx, _nodes.size())) {
```

Comment:
Here, we can replace this check with `if (idx == -1 || size_t(idx) >= _nodes.size())`.

Reply:
i am not in favor of this change. because it hardcodes the knowledge of "idx should be -1, 0 or a positive number" in an obscure way. IMHO, greater equal is more readable.